uc.c

```diff
@@ -1036,6 +1036,16 @@ uc_err uc_hook_add(uc_engine *uc, uc_hook *hh, int type, void *callback,
         hook->insn = va_arg(valist, int);
         va_end(valist);
 
+
+#ifdef UNICORN_HAS_X86
```

Comment:
would be better if we move everything out, so there is no more "ifdef UNICORN_HAS_X86" in uc.c.

you can think of adding a new function pointer to `uc_struct`. see `include/priv.h` for examples.